### PR TITLE
Progress on UI and docs

### PR DESCRIPTION
**Pull request by zehicle**
Frame out UI work for @Meshiest efforts

Add to Docs based on work to support links from UI

### CODE REVIEW

**Comment by galthaus** [comment]
This seems to need rebasing to be clean.

**Comment by zehicle** [comment]
@galthaus fixed the rebase.  sorry about that!

File: doc/ui.rst [hunk 1]
```diff
@@ -9,12 +9,40 @@
...
 
-There is no UI at this time.
+The Rocket Skates UI is intended to help users with basic operational needs.  Advanced users should review the API documentation or consider using Digital Rebar for integration and workflow capabilities.
 
+.. _rs_ui_subnets:
+
+Subnets
+-------
+
+Configurating Subnets is a critical first step in Rocket Skates operation.  The basic UI will show all configurated subnets and provide an easy way to add broadcast subnets based on the known interfaces.
```
Comment (VictorLowther) [inline doc/ui.rst]: Configurating?  Really?

File: doc/ui.rst [hunk 2]
```diff
@@ -9,12 +9,40 @@
...
+
+To edit or delete a subnet, click on the name of the subnet to populate the editing area below the list.  To create a relay subnet, click on the add subnet link.  To create a broadcast subnet, click on the link provided after the name of the unassigned interfaces.
+
+There are two primary types of subnets: broadcast and relay:
+
+  * Broadcast subnets are associated with the addresses provided by the Rocket Skates host system.  Rocket Skates can have multiple DHCP ranges; however, you can only assign one address per interface _and_ those ranges must include the IP of the interface.  By convention, subnets associated with an interface will be named as the interface.
+  * Relay subnets answer requests forwarded from DHCP relays such as those provided by switches.  These can be any suitable IP range.  Since the Relay subnets are not broadcast, they do not conflict with existing DHCP servers in the environment.s
```
Comment (VictorLowther) [inline doc/ui.rst]: Extra.s at the end.s

File: doc/ui.rst [hunk 3]
```diff
@@ -9,12 +9,40 @@
...
+
+Rocket Skates can operate in a permissive reservation more or require users to whitelist systems before they are serviced.  The `OnlyReservations` flag will operate as a reservations required (whitelist) mode when true; otherwise, Rocket Skates will give out addresses to any valid DHCP request.
```
Comment (VictorLowther) [inline doc/ui.rst]: Permissive reservation mode.

File: doc/ui.rst [hunk 4]
```diff
@@ -9,12 +9,40 @@
...
+
+In additionl to serving IPs, DHCP servers provide critical confinguration (aka `DHCP Options <https://en.wikipedia.org/wiki/Dynamic_Host_Configuration_Protocol#DHCP_options>`_) information to the clients.  Setting Option 67, Next Boot, is essential for Rocket Skates to operate as a Provisioner.  This information includes next boot (67), gateway (3), domain name (15), DNS (6) and other important information.  It is encoded in the responses according to `IETF RFC 2132 <https://tools.ietf.org/html/rfc2132>`_
+
+Consult to `Godocs <https://godoc.org/github.com/rackn/rocket-skates/backend#Subnet>`_ for more details about the specific fields.
```
Comment (VictorLowther) [inline doc/ui.rst]: Consult the

File: midlayer/dhcp_server.go [hunk 5]
```diff
@@ -10,6 +10,7 @@ import (
 
 func RunDhcpHandler(dhcpInfo *backend.DataTracker, ifs []*backend.Interface) {
 	handlers := make(map[int]dhcp.Handler, 0)
+	log.Println("DOCUMENTATION http://rocket-skates.readthedocs.io/en/latest/doc/faq-troubleshooting.html")
```
Comment (VictorLowther) [inline midlayer/dhcp_server.go]: No raw use of the log package, please.  Use dhcpInfo.Logger.Println instead.

Comment (zehicle) [inline midlayer/dhcp_server.go]: I also updated the println on the "starting interface lines"

**Comment by zehicle** [comment]
@VictorLowther addressed issues identified

File: doc/ui.rst [hunk 6]
```diff
@@ -9,12 +9,40 @@
...
 
-There is no UI at this time.
+The Rocket Skates UI is intended to help users with basic operational needs.  Advanced users should review the API documentation or consider using Digital Rebar for integration and workflow capabilities.
 
+.. _rs_ui_subnets:
+
+Subnets
+-------
+
+Configuring Subnets is a critical first step in Rocket Skates operation.  The basic UI will show all configurated subnets and provide an easy way to add broadcast subnets based on the known interfaces.
```
Comment (VictorLowther) [inline doc/ui.rst]: Still missed a configurate

File: doc/ui.rst [hunk 7]
```diff
@@ -9,12 +9,40 @@
...
+
+To edit or delete a subnet, click on the name of the subnet to populate the editing area below the list.  To create a relay subnet, click on the add subnet link.  To create a broadcast subnet, click on the link provided after the name of the unassigned interfaces.
```
Comment (VictorLowther) [inline doc/ui.rst]: These instructions should probably come after explaining the difference between broadcast subnets and relay subnets.

File: doc/ui.rst [hunk 8]
```diff
@@ -9,12 +9,40 @@
...
+
+There are two primary types of subnets: broadcast and relay:
+
+  * Broadcast subnets are associated with the addresses provided by the Rocket Skates host system.  Rocket Skates can have multiple DHCP ranges; however, you can only assign one address per interface _and_ those ranges must include the IP of the interface.  By convention, subnets associated with an interface will be named as the interface.
```
Comment (VictorLowther) [inline doc/ui.rst]: One IP address per interface (which is not under our control), or one DHCP subnet per interface?

File: doc/ui.rst [hunk 9]
```diff
@@ -9,12 +9,40 @@
...
+  * Relay subnets answer requests forwarded from DHCP relays such as those provided by switches.  These can be any suitable IP range.  Since the Relay subnets are not broadcast, they do not conflict with existing DHCP servers in the environment.
+
+Rocket Skates can operate in a permissive reservation more or require users to whitelist systems before they are serviced.  The `OnlyReservations` flag will operate as a reservations required (whitelist) mode when true; otherwise, Rocket Skates permissive reservation mode will give out addresses to any valid DHCP request.
```
Comment (VictorLowther) [inline doc/ui.rst]: Still have a more instead of a mode.

Comment (zehicle) [inline doc/ui.rst]: fixed

**Pull request merged**